R/make-traceability-matrix.R

```diff
@@ -270,8 +276,19 @@ map_tests_to_functions <- function(exports_df, pkg_source_path, verbose){
 
   # map over test files and parse all functions called in those files
   func_test_df <- purrr::map_dfr(test_files, function(test_file.i) {
-    parsed_df <- test_file.i %>%
-      parse(keep.source = TRUE) %>%
+
+    test_dir.i <- fs::path_rel(dirname(test_file.i), pkg_source_path)
+
+    exprs <- tryCatch(parse(test_file.i, keep.source = TRUE), error = identity)
+    if (inherits(exprs, "error")) {
+      warning("Failed to parse ", test_file.i, ": ", conditionMessage(exprs))
+      # assign NA to ensure the test_file still gets captured at this stage
+      # TODO: do we want specific handling for NA cases? (besides the warning)
+      # Maybe append a separate list of un-mapped test files? Figure out a different way
```

Comment:
Given this [comment](https://github.com/metrumresearchgroup/mpn.scorecard/issues/46#issuecomment-1724406122), I will just delete these `TODO` comments. 

The `NA_character` assignment (vs an empty `character()`) doesn't really matter since it will be filtered to just exports at the end. However, it is meant to ensure the `test_file` was captured/added to the list before left joining onto the exports, which IMO does provide _some_ value.

As an aside, it could be worth refactoring this function to use `get_toplevel_assignments` (which would require its own refactor to not look in the R directory)

Reply:
> As an aside, it could be worth refactoring this function to use `get_toplevel_assignments` (which would require its own refactor to not look in the R directory)

I don't follow. `get_toplevel_assignments` is looking for top-level assignments while this function `map_tests_to_functions` looks for symbols and function calls.